examples/lua_REPL/repl.lua

```diff
@@ -0,0 +1,91 @@
+--[[
+   @file repl.lua
+   @brief   Read-eval-print loop for LUA
+   @author  Juan Carrano <j.carrano@fu-berlin.de>
+]]
+
+local _R_EVAL = 0
+local _R_CONT = 1
+local _R_EXIT = 2
+local _R_ERROR = 3
+
+--[[ Read code from standard input (whatever stdin means for lua)
+    @return action_code     what the eval loop should do
+    @return code_or_msg     either code (_R_EVAL) or a message (_R_ERROR) or nil
+                            (_R_CONT, _R_EXIT).
+]]
+
```

Comment:
Could you make a few changes to make this easier to understand out-of-the-box - i.e. so that someone can come in with no help and learn a few lua features? Suggestions are:
- rename "maybe_code" to something like "is_executable_code" or something and explain that the three `if maybe_code == nil` sections check whether it's an expression, then a statement, then a line from a multiline
- briefly explain the coroutine functionality
- explain what the typical dual return codes for lua are for
- maybe explain that load() compiles chunks line by line and then pcall() executes the compiled bytecode  

I understand that comments come at a memory cost since it's all being parsed into the array. Would be good to find a good balance here. I think if you can go as far as possible with readable code itself that would be good, by renaming some of the variables. Potentially, could also write some notes in the readme?

Reply:
I agree that it's poorly documented. However, teaching Lua is beyond the scope of our documentation. I would expect people who want to use lua to learn the core language elsewhere, and in RIOT we can limit ourselves to document only the RIOT-specific extensions and usage.

Reply:
ok cool, thanks for making the changes